File: frontend/src/types/certificate.ts
export interface MatchedCertificate {
  id: number;
  serial_number: string;
  common_name: string;
  sans: string[];
  issuer: string;
  not_before: string; // ISO 8601
  not_after: string; // ISO 8601
  keyword_id: number;
  keyword_value: string;
  matched_domain: string;
  ct_log_index: number;
  discovered_at: string; // ISO 8601
}

export interface CertificatesResponse {
  certificates: MatchedCertificate[];
  total: number;
  page: number;
  per_page: number;
}

File: backend/internal/model/certificate.go
package model

import "time"

type MatchedCertificate struct {
	ID            int       `json:"id"`
	SerialNumber  string    `json:"serial_number"`
	CommonName    string    `json:"common_name"`
	SANs          []string  `json:"sans"`
	Issuer        string    `json:"issuer"`
	NotBefore     time.Time `json:"not_before"`
	NotAfter      time.Time `json:"not_after"`
	KeywordID     int       `json:"keyword_id"`
	KeywordValue  string    `json:"keyword_value,omitempty"`
	MatchedDomain string    `json:"matched_domain"`
	CTLogIndex    int64     `json:"ct_log_index"`
	DiscoveredAt  time.Time `json:"discovered_at"`
}
